sources/driverdog/src/main.rs

```diff
@@ -125,35 +141,37 @@ fn link_modules<S>(driver_config: &DriverConfig, kernel_version: S) -> Result<()
 where
     S: AsRef<str>,
 {
-    let kernel_version = kernel_version.as_ref();
-    // The directory with the module's objects
-    let driver_path = Path::new(&driver_config.objects_source).to_path_buf();
-    // Destination for the kernel modules
-    let modules_path = Path::new(LIB_MODULES_PATH)
-        .join(kernel_version)
-        .join(&driver_config.lib_modules_path);
-    // Directory to store temp artifacts
-    let build_dir = tempfile::tempdir().context(error::TmpDirSnafu)?;
-    // This script is used to link the kernel module
-    let common_module_script = Path::new(KERNEL_SOURCES)
-        .join(kernel_version)
-        .join("scripts/module.lds");
-
-    // First, link the object files, and store them in the temp directory
-    for (name, object_file) in driver_config.object_files.iter() {
-        link_object_file(name, object_file, &build_dir, &driver_path)?;
-    }
+    if let Some(objects_source) = &driver_config.objects_source {
```

Comment:
I'd prefer to do this test before calling the function, ideally by matching on the type (if we have different `Copyable` and `Linkable` types).